petastorm/spark/spark_dataset_converter.py

```diff
@@ -0,0 +1,310 @@
+#  Copyright (c) 2020 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import atexit
+import shutil
+import threading
+import uuid
+import logging
+import warnings
+
+from pyarrow import LocalFileSystem
+from pyspark.sql.session import SparkSession
+from six.moves.urllib.parse import urlparse
+
+from petastorm import make_batch_reader
+from petastorm.fs_utils import FilesystemResolver
+
+DEFAULT_ROW_GROUP_SIZE_BYTES = 32 * 1024 * 1024
+
+
+def _get_spark_session():
+    return SparkSession.builder.getOrCreate()
+
+
+_parent_cache_dir_url = None
+
+
+def get_parent_cache_dir_url():
+    """
+    Get parent cache dir url from `petastorm.spark.converter.defaultCacheDirUrl`
+    We can only set the url config once.
+    """
+    global _parent_cache_dir_url  # pylint: disable=global-statement
+
+    conf_url = _get_spark_session().conf \
+        .get("petastorm.spark.converter.defaultCacheDirUrl", None)
+
+    if conf_url is None:
+        raise ValueError(
+            "Please specify the parameter cache_dir_url denoting the parent "
+            "directory to store intermediate files, or set the spark config "
+            "`petastorm.spark.converter.defaultCacheDirUrl`.")
+
+    if _parent_cache_dir_url is not None:
+        if _parent_cache_dir_url != conf_url:
+            raise RuntimeError(
+                "petastorm.spark.converter.cacheDirUrl has been set to be "
```

Comment:
`petastorm.spark.converter.cacheDirUrl`->`petastorm.spark.converter.defaultCacheDirUrl`. Here and elsewhere.

Reply:
What about change the conf name to be 
`petastorm.spark.converter.parentCacheDirUrl` ? There's no default concept on this, because each converter have to use this conf, no private setting.